### ISSUE TITLE
spawn UNKNOWN

### ISSUE THREAD

**Issue by franck-serot** [mantoni/mochify.js#115]
Hi,
I try to use mochify and grunt-mochify ("grunt-mochify": "^0.2.0").
I could run a simple test with mochify but I have an error with the grunt-mochify : spawn UNKNOWN
You could find the log produced with verbose grunt option.
Would you please help me to understand this problme and solve it ?
Best regards,
Franck

Case OK
---
node_modules\\.bin\mochify ./app/client/test/scripts/main.spec.js
phantomjs:
  0 passing (7ms)
  1 failing

  1) main main should return true:
     AssertionError: main should return true: expected false to be true
      at js/bundle:247
      at js/bundle:595
      at addProperty (js/bundle:3938)

Error: Exit 1
Gruntfile.js
--------------
...
        client : {
		root : 'app/client',
		dist : 'dist/client'
	},
       mochify: {
	    	test: {
	    	  options: {
	            reporter: 'xunit',
			    captureFile: '<%=client.dist%>/test/test-reports.xml',
			  },
	          src: [ '<%=client.root%>/test/scripts/**/*.spec.js']
	        }
	   }

log
---
...
Running "mochify" task

Running "mochify:test" (mochify) task
Verifying property mochify.test exists in config...OK
Files: app/client/test/scripts/main.spec.js
Options: reporter="xunit", colors=false, debug=false, captureFile="dist/client/test/test-reports.xml"
Spawning Mochify with command D:\workspace\myProject\node_modules\mochify\bin\cmd.js
        and arguments: --reporter=xunit, --colors=false, --debug=false, --captureFile=dist/client/test/test-reports.xml, app/client/test/scripts/main.spec.js
Warning: spawn UNKNOWN Use --force to continue.

Aborted due to warnings.

**Comment by franck-serot** [comment]
Hi,
I think I find an explanation and a workaround of this problem.
In add, I add a new option in order to redirect the output from stdout to a file.
Could you please confirm or infirm the explanation and the workaround ?
Best regards.
Franck

mochify.js
---
```javascript
'use strict';

var spawn = require('child-process-promise').spawn;
var map = require('lodash.map');

function mochifyTask(grunt) {

  grunt.registerMultiTask('mochify', 'Run Mochify', function (target) {
    var done = this.async();

    var options = this.options({
      reporter: 'spec',
      colors: !!process.stdout.isTTY,
      debug: target === 'debug'
    });

    // new option to redirect output from stdout to a file
    var output = options.output || undefined;
    delete options.output;
    
    var args = map(options, function (value, name) {
      return '--' + name + '=' + value;
    }).concat(this.filesSrc);
    
    // The script mochify/bin/cmd shall be run with D:\bin\nodejs_node.exe whereas the script file .bin/mochify.cmd could be called directly.
    //var mochifyPath = require.resolve('mochify/bin/cmd');
    var mochifyPath = require.resolve('.bin/mochify.cmd');

    grunt.verbose.writeln('Spawning Mochify with command %s\n\tand ' +
      'arguments: %s', mochifyPath, grunt.log.wordlist(args));

    spawn(mochifyPath, args)
      .progress(function (childProcess) {
        childProcess.stdout.on('data', function (data) {
          // if the 'output' option exists then redirect output from stdout to a file
          if (output != undefined) {
              var fs = require('fs');
              var wstream = fs.createWriteStream(output);
              wstream.write(String(data));
              wstream.end();
          } else {
              console.log(String(data));
          }
        });
        childProcess.stderr.on('data', function (data) {
          console.error(String(data));
        });
      })
      .then(done)
      .fail(function (err) {
        grunt.fail.warn(err);
      });
  });

}

module.exports = mochifyTask;
```

**Comment by mantoni** [comment]
If this is an issue with `grunt-mochify`, please report [over there](https://github.com/decipherinc/grunt-mochify). I know nothing about grunt.

**Comment by boneskull** [comment]
@franck-serot I released a new version of grunt-mochify just now; please see if it solves your issue.

**Comment by franck-serot** [comment]
I tested the new version of grunt-mochify and 'mochify' is now run.
But with the option 'debug: false', the inspector is opened.
I think it's a minor bug but some cleaning is required in the argument list.
In add, I tested my workaround to redirect the ouput of 'mochify' from stdout to a file but the file is corrupted.
We should use the option 'output' of 'mochify' but this option is not reachable.
